composer/trainer/trainer.py

```diff
@@ -2140,12 +2140,155 @@ def predict_batch_end(self, state: State, logger: Logger) -> None:
+        either the eval() method or during training init().
+
+        Examples:
+        .. testcode::
+
+            trainer = Trainer(
+                model=model,
+                train_dataloader=train_dataloader,
+                max_duration="2ep",
+                device="cpu",
+            )
+
+            trainer.fit()
+
+            # run eval
+            trainer.eval(
+                eval_dataloader=eval_dataloader,
+            )
+
+        Or, if the ``eval_dataloader`` is provided during init:
+
+        .. testcode::
+
+            trainer = Trainer(
+                model=model,
+                eval_dataloader=eval_dataloader,
+                train_dataloader=train_dataloader,
+                max_duration="2ep",
+                device="cpu",
+            )
+
+            trainer.fit()
+
+            # eval_dataloader already provided:
+            trainer.eval()
+
+        For multiple metrics or dataloaders, use :class:`.Evaluator` to provide
+        identifier names. For example, to run the GLUE task:
+
+        .. code:: python
+
+            from composer.core import Evaluator
+            from composer.models.nlp_metrics import BinaryF1Score
+
+            glue_mrpc_task = Evaluator(
+                label='glue_mrpc',
+                dataloader=mrpc_dataloader,
+                metric_names=['BinaryF1Score', 'Accuracy']
+            )
+
+            glue_mnli_task = Evaluator(
+                label='glue_mnli',
+                dataloader=mnli_dataloader,
+                metric_names=['Accuracy']
+            )
+
+            trainer = Trainer(
+                ...,
+                eval_dataloader=[glue_mrpc_task, glue_mnli_task],
+                ...
+            )
+
+        The metrics used are defined in your model's ``get_metrics()`` method. For more information,
+        see :doc:`/trainer/evaluation`.
+
+        .. note::
+
+            This eval API was recently changed to better much the trainer fit API. Please migrate your
+            code to using the new design here. For backwards compatibility, the old API can still be
+            invoked by calling ``_eval_loop()``, however this is not recommended as this may be
+            removed in the future.
+
+        Args:
+            eval_dataloader (DataLoader | DataSpec | Evaluator | Sequence[Evaluator], optional): Dataloaders
+                for evaluation.  If not provided, defaults to using the
+                ``eval_dataloader`` provided to the trainer init().
+            subset_num_batches (int, optional): Evaluate on this many batches. Default to ``-1`` (the entire
+                dataloader. Can also be provided in the trainer init()as ``eval_subset_num_batches``.
+            dataloader: Removed, do not use.
+            dataloader_label: Removed, do not use.
+            metrics: Removed, do not use.
+            log_level: Removed, do not use.
+
+        """
+        if any([dataloader, dataloader_label, metrics]):
```

Comment:
```suggestion
        if any([k in kwargs for k in ['dataloader', 'dataloader_label', 'metrics', 'log_level']]):
```
see above comment for context. Basically, we have a catch-all kwargs that we check for deprecated arguments.